### PR TITLE
Add new extension point for repository avatars

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Add extension points to show custom repository avatars. Also enhance the file upload component.

### Your checklist for this pull request

- [x] PR is well described and the description can be used as commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [ ] New code is covered with unit tests
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog`
- [x] Definition of Done's fulfilled: [DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/definition-of-done.md) // [UI DoD](https://github.com/scm-manager/scm-manager/blob/develop/docs/en/development/ui-dod.md)
- [ ] Documentation updated (only necessary for new features or changed behaviour)

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

**Comment by pfeuffer** [comment]
Somewhere there was a documentation for our extension points, wasn't it?

File: scm-ui/ui-webapp/src/repos/containers/EditRepo.tsx [hunk 1]
```diff
@@ -57,6 +57,7 @@ const EditRepo: FC<Props> = ({ repository }) => {
       <Subtitle subtitle={t("repositoryForm.subtitle")} />
       <ErrorNotification error={error} />
       <RepositoryForm repository={repository} loading={isLoading} modifyRepository={update} />
+      <ExtensionPoint name="repo-config.details" props={extensionProps} />
```
Comment (pfeuffer) [inline scm-ui/ui-webapp/src/repos/containers/EditRepo.tsx:60]: Do we need a `renderAll={true}` here?

Comment (eheimbuch) [reply scm-ui/ui-webapp/src/repos/containers/EditRepo.tsx:60]: Why not? There could be other plugins which want to provide configuration forms to the repo settings page?

Comment (pfeuffer) [reply scm-ui/ui-webapp/src/repos/containers/EditRepo.tsx:60]: Ah, and please fix the storyshots.

**Comment by pfeuffer** [comment]
Ah, and please fix the storyshots.

**Pull request merged**